protocols/v2/framing-sv2/src/framing2.rs

```diff
@@ -82,20 +89,19 @@ impl<T, B> Default for Sv2Frame<T, B> {
     }
 }
 
+/// Abstraction for a Noise Frame
+/// In practice, it is only used as the alias `HandShakeFrame`
+/// Contains only a `Slice` payload
 #[derive(Debug)]
 pub struct NoiseFrame {
     payload: Slice,
 }
 
+/// The only practical usage of `NoiseFrame`
 pub type HandShakeFrame = NoiseFrame;
 
-#[cfg(feature = "with_buffer_pool")]
-impl<A> From<EitherFrame<A, Vec<u8>>> for Sv2Frame<A, buffer_sv2::Slice> {
-    fn from(_: EitherFrame<A, Vec<u8>>) -> Self {
-        unreachable!()
-    }
-}
 impl NoiseFrame {
+    /// Returns payload of `NoiseFrame` as a `Vec<u8>`
```

Comment:
As already said above I would always call it HandShakeFrame for now (then we will remove it entirely).